pkgs/by-name/gl/globalprotect-openconnect_2/package.nix

```diff
@@ -0,0 +1,134 @@
+{
+  perl,
+  jq,
+  fetchFromGitHub,
+  lib,
+  openconnect,
+  libsoup,
+  webkitgtk_4_0,
+  pkg-config,
+  callPackage,
+  rustPlatform,
+  glib,
+  atk,
+  gdk-pixbuf,
+  pango,
+  cairo,
+  harfbuzz,
+  gtk3,
+  zlib,
+  makeDesktopItem,
+  includeUnfreeGui ? false,
+}:
+let
+  version = "2.3.7";
+  pname = "globalprotect-openconnect";
+  desktop = makeDesktopItem {
+    name = "GlobalProtect Openconnect VPN Client";
+    desktopName = "GlobalProtect Openconnect VPN Client";
+    comment = "A GUI for GlobalProtect VPN";
+    genericName = "GlobalProtect VPN Client";
+    exec = "gpclient launch-gui %u";
+    icon = "gpgui";
+    categories = [
+      "Network"
+      "Dialup"
+    ];
+    mimeTypes = [ "x-scheme-handler/globalprotectcallback" ];
+  };
+
+  gpgui = if includeUnfreeGui then (callPackage ./gui.nix { }) else null;
+
+in
+rustPlatform.buildRustPackage {
+  inherit version pname;
+
+  src = fetchFromGitHub {
+    owner = "yuezk";
+    repo = "GlobalProtect-openconnect";
+    rev = "v${version}";
+    hash = "sha256-Zr888II65bUjrbStZfD0AYCXKY6VdKVJHQhbKwaY3is=";
+  };
+
+  nativeBuildInputs = [
+    perl
+    jq
+    openconnect
+    libsoup
+    webkitgtk_4_0
+    pkg-config
+  ];
+
+  PKG_CONFIG_PATH = lib.strings.concatMapStringsSep ":" (pkg: "${pkg}/lib/pkgconfig/") [
+    glib.dev
+    libsoup.dev
+    webkitgtk_4_0.dev
+    atk.dev
+    gdk-pixbuf.dev
+    pango.dev
+    cairo.dev
+    harfbuzz.dev
+    gtk3.dev
+    openconnect
+    zlib
+  ];
+
+  NIX_CFLAGS_COMPILE = (map (pkg: "-I${pkg}/include") [ openconnect.dev ]);
+
+  NIX_CFLAGS_LINK = (
+    map (pkg: "-L${lib.getLib pkg}/lib") [
+      openconnect
+      zlib
+    ]
+  );
+
+  cargoHash = "sha256-cdhhBUQASrnfjeJxkwx39vr/KHeQlBh0wMvw+Q7EK98=";
+
+  postPatch =
+    let
+      replacements = builtins.filter (v: v != null) [
+        "/usr/bin/gpclient $out/bin/gpclient"
+        "/usr/bin/gpservice $out/bin/gpservice"
+        "/usr/bin/gpgui-helper $out/bin/gpgui-helper"
+        "/usr/bin/gpauth $out/bin/gpauth"
+        (if includeUnfreeGui then "/usr/bin/gpgui ${gpgui}/bin/gpgui" else null)
+      ];
+      replacements-string = lib.strings.concatMapStringsSep " " (l: "--replace-fail ${l}") replacements;
+    in
+    ''
+      substituteInPlace crates/gpapi/src/lib.rs ${replacements-string}
+    '';
```

Comment:
Need to patch the `vpnc-scripts` reference here too.

https://github.com/NixOS/nixpkgs/blob/76612b17c0ce71689921ca12d9ffdc9c23ce40b2/pkgs/by-name/gp/gpclient/package.nix#L34-L35

Reply:
Can you also make your `substituteInPlace` usage use `--replace-fail` so anyone using upgrade automation will know to check if something has changed if the substitute no longer is needed or does not work?

Reply:
Regarding vpnc-script patch, it is useless patching it since these references only works with the gui we are not building here sadly 😢

Reply:
Replace fail is used in the replacement string already.

Reply:
I fixed it anyway for the cli version.